src/globus_sdk/experimental/auth_requirements_error/utils.py

```diff
@@ -0,0 +1,132 @@
+from __future__ import annotations
```

Comment:
I would like us to not call this module `utils` because I think it's non-descriptive and implies that it's okay for us to have some loosely organized utilities which we then expose to users.
It also enters the same territory as `globus_sdk.utils` which is very explicitly declared as non-public.

In #796 I went with `functional_interface`, but I'm also okay with `handlers`, `helpers`, etc.

I just want to avoid this _particular_ name because it's a name we already are using in a broader context for a different purpose.

(I prefer my own suggestion of `functional_interface` because it declares that these methods, in the very module name, are specifically public-facing.)